File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    first = [1, 2, 3, 4, 5]
    second = [2, 1, 2, 3, 2, 4, 2, 5]
    third = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    
    scores = [0, 0, 0]
    for i in range(len(answers)):
        
        
        if first[i % len(first)] == answers[i]:
            scores[0] += 1
        
        if second[i % len(second)] == answers[i]:
            scores[1] += 1
        
        if third[i % len(third)] == answers[i]:
            scores[2] += 1
    
    max_score = max(scores)
    
    answer = []
    
    for i in range(3):
        if max_score == scores[i]:
            answer.append(i+1)
    
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;

class Solution {
    public int[] solution(int[] answers) {
        int[] count = new int[3];
        
        int[] first = {1,2,3,4,5};
        int[] second = {2,1,2,3,2,4,2,5};
        int[] third = {3,3,1,1,2,2,4,4,5,5};
        
        for(int i=0; i<answers.length; i++) {
            if(first[i%5] == answers[i]) {
                count[0] += 1;
            }
            if(second[i%8] == answers[i]) {
                count[1] += 1;
            }
            if(third[i%10] == answers[i]) {
                count[2] += 1;
            }
        }
        
        int maxScore = 0;
        for(int score: count) {
            if(maxScore < score) {
                maxScore = score;
            }
        }
        
        ArrayList<Integer> answer = new ArrayList<>();
        for(int i=0; i<3; i++) {
            if(maxScore == count[i]) {
                answer.add(i+1);
            }
        }
        
        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}